fiftyone/core/odm/database.py

```diff
@@ -1484,6 +1484,90 @@ def delete_runs(name, dry_run=False):
     )
 
 
+def get_indexed_values(
+    collection,
+    field_or_fields,
+    *,
+    index_key=None,
+    query=None,
+    values_only=False,
+):
+    """Returns the values of the field(s) for all samples in the given collection
+    that are covered by the index. Raises an error if the field is not indexed.
+
+    Args:
+        collection: a ``pymongo.collection.Collection`` or
+            ``motor.motor_asyncio.AsyncIOMotorCollection``
+        field_or_fields (str or list): the field name(s) to retrieve.
+        index_key (None): the name of the index to use. If None, the default
+            index name will be constructed from the field name(s).
+        query (dict, optional): a selection filter to apply when querying.
+            For performance, this should only include fields that are in
+            the specified index.
+        values_only (False): whether to remove field names from the resulting list.
+            If True, the field names are removed and only the values will be
+            returned as a list for each sample. If False, the field names are
+            preserved and the values will be returned as a dict for each sample.
+    Returns:
+        a list of values for the specified field or index keys for each sample
+        sorted in the same order as the index
+    """
+    try:
+        cursor = _iter_indexed_values(
+            collection, field_or_fields, index_key=index_key, query=query
+        )
+
+        if field_or_fields == "id":
```

Comment:
Noting a minor case like `ds.values(["id", "filepath"])` where we still want to cast "_id" to string.
Not worth doing anything about it here though.